### PR TITLE
News app

### PR DESCRIPTION
**Pull request by GlebPilipenko**
news_app - did news_app, adaptive, carousel, .env

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,4 +23,7 @@ yarn-debug.log*
 yarn-error.log*
 
 # idea
-.idea
\ No newline at end of file
+.idea
+
+# .env
+.env
```
Comment (ghost) [inline .gitignore]: add `.env.example` file with 
```
REACT_APP_NEWS_BASE_URL =
REACT_APP_API_KEY = 
```

File: public/index.html [hunk 2]
```diff
@@ -36,9 +36,8 @@
   To begin the development, run `npm start` or `yarn start`.
   To create a production bundle, use `npm run build` or `yarn build`.
 -->
-<div data-city="Minsk"
-     data-app="weather-widget"
-     data-coordinates="53.8847208, 27.5233291">
-</div>
+<!--<div data-city="Minsk"-->
+<!--     data-app="weather-widget" data-coordinates="53.8847208, 27.5233291"></div>-->
+<div data-city="Minsk" data-app="news-widget" data-country="Germany"></div>
```
Comment (ghost) [inline public/index.html]: I can see the news after removing data-city and data-country...

Comment (ghost) [reply public/index.html]: We'll need to see news about the city or the country (if there are no news about the city)

File: public/manifest.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "short_name": "React App",
   "name": "Create React App Sample",
+  "icons": [],
```
Comment (ghost) [inline public/manifest.json]: ?

File: src/App.tsx [hunk 4]
```diff
@@ -2,19 +2,31 @@ import React, {useEffect, useState} from 'react';
 import {Weather} from './components/weather/Weather';
```
Comment (ghost) [inline src/App.tsx:2]: image

File: src/api/newsAPI.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+const instance = axios.create({
+    baseURL: `${process.env.REACT_APP_NEWS_BASE_URL}`,
+});
+
+export const newsAPI = {
+    getNewsCity(city: string | undefined) {
+        return instance.get<NewsAPIType>(
+            `everything?q=${city}&pageSize=5&apiKey=${process.env.REACT_APP_API_KEY}`
```
Comment (ghost) [inline src/api/newsAPI.ts]:
```
everything?q=${city}&pageSize=5&apiKey=${process.env.REACT_APP_API_KEY}
everything?q=${country}&pageSize=5&apiKey=${process.env.REACT_APP_API_KEY}
```

move to the const with an argument query (city or country)

File: src/api/newsAPI.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+const instance = axios.create({
```
Comment (ghost) [inline src/api/newsAPI.ts]:
```
const newsApi = axios.create({ ... });
const getNewsCity = newsApi.get();
```

File: src/api/newsAPI.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    getNewsCountry(country: string | undefined) {
+        return instance.get<NewsAPIType>(
+            `everything?q=${country}&pageSize=5&apiKey=${process.env.REACT_APP_API_KEY}`
+        )
+            .then(countryResult => countryResult);
+    }
+};
+
+type SourceType = {
```
Comment (ghost) [inline src/api/newsAPI.ts:19]: all types should be at the top of the file or better in the separate file

File: src/api/weatherAPI.ts [hunk 8]
```diff
@@ -12,6 +12,11 @@ export const weatherAPI = {
     }
 };
 
+type CoordType = {
+    lon: number
```
Comment (ghost) [inline src/api/weatherAPI.ts]: `;`

File: src/components/news/News.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+import style from './News.module.css';
+
+type PropsType = {
+    city?: string;
+    country?: string;
+}
+
+export const News: React.FC<PropsType> = (props) => {
+
+    const [state, setState] = useState<NewsAPIType>({
```
Comment (ghost) [inline src/components/news/News.tsx]: const initialState

**Review by alexwerst — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 10]
```diff
@@ -1,20 +1,30 @@
...
-    enum WidgetTitles {
-        WeatherWidget = 'weather-widget',
-    };
-
-    return <div className={s.wrapper}>
-        {props.app === WidgetTitles.WeatherWidget &&
-        <Weather city={props.city} />}
-    </div>;
-};
+export const App: React.FC<PropsType> = (props) => (
```
Comment (aashihmin) [inline src/App.tsx]: App should return necessary component only (according to the props.app). Leave switch() only

File: src/components/common/errorMessage/ErrorMessage.tsx [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import style from './ErrorMessage.module.css';
+
+type PropsType = {
+    errorMessage: string
+}
+
+export const ErrorMessage: React.FC<PropsType> = ({errorMessage}) => {
```
Comment (aashihmin) [inline src/components/common/errorMessage/ErrorMessage.tsx]:
```
export const ErrorMessage: React.FC<PropsType> = ({errorMessage}) => (
    <div className={style.error__container}>
        <div className={style.error__block}>
            {errorMessage}
        </div>
   </div>
);
```

File: src/utils/GetFormatedDate.ts [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+    const date = new Date(+dt * oneThousand);
+    const mm = date.getMonth() + 1;
+    const yyyy = date.getFullYear();
+    const dd = date.getDate();
+
+    return `${mm}/${dd}/${yyyy}`;
+};
+
+export const getDateToTimeStamp = (strDate: string) => {
+    return Date.parse(strDate) / 1000;
```
Comment (aashihmin) [inline src/utils/GetFormatedDate.ts]: magic number

**Review by aashihmin — CHANGES_REQUESTED** [review]

**Review by aashihmin — APPROVED** [review]

**Review by alexwerst — APPROVED** [review]

**Pull request merged**